### DISCUSSION TITLE
What I want to make a character status popup?

### DISCUSSION THREAD

**Discussion by locus84** [LibraStack/UnityMvvmToolkit#70] [Q&A]
I want to set initial parameter for the viewmodel that is made for a character status.
but it looks like I can't feed my own viewmodel to a CanvasView as it requests the viewmodel when awake.
I'm new to mvvm so teach me how to. thank you in advance.

**Comment by ChebanovDD** [comment]
You can override the base viewmodel instance creation. [Docs](https://github.com/LibraStack/UnityMvvmToolkit?tab=readme-ov-file#canvasviewtbindingcontext).

```csharp
public class CounterView : CanvasView<CounterViewModel>
{
    // Override the base viewmodel instance creation.
    // Required in case the viewmodel doesn't have a parameterless constructor.
    protected override CounterViewModel GetBindingContext()
    {
        return new CounterViewModel(69);
    }
}
```

**Reply by locus84** [reply]
Thanks! I saw docs and also checked source code.
I initially thought I can pass some parameters using ctor of CounterViewModel itself as you suggested.
But after digging up VContainer and the MvvmToolkit sample, it looks like I need some kind of Broker or through DI container.

```
public class CounterView : CanvasView<CounterViewModel>
{
    // Override the base viewmodel instance creation.
    // Required in case the viewmodel doesn't have a parameterless constructor.
    protected override CounterViewModel GetBindingContext()
    {
        return _scopedContext.Resolve<CounterViewModel>();
       //or
        return new CounterViewModel(_scopedContext.Resolve<ICounterBroker>().value);
    }
}
```
